src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -791,20 +792,19 @@ static int rbd_nbd(int argc, const char *argv[])
 
   argv_to_vec(argc, argv, args);
   env_to_vec(args);
-  auto cct = global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT,
-			 CODE_ENVIRONMENT_DAEMON,
-			 CINIT_FLAG_UNPRIVILEGED_DAEMON_DEFAULTS);
-  g_ceph_context->_conf->set_val_or_die("pid_file", "");
+
+  vector<const char*> temp_args(args);
+  md_config_t().parse_argv(temp_args);
 
```

Comment:
@trociny  two questions:
1) do you mean we would better move argc and argv into do_map instead of args, and we call argv_to_vec again in do_map?

2) is this env_to_vec unnecessary? I want to get more info about this. Do you mean Just remove it, or move it to do_map? If Just remove it, I prefer to add a seperate commit for this removal in this pr.